dmutils/errors/frontend.py

```diff
@@ -10,7 +12,26 @@ def csrf_handler(csrf_error):
     :param logger: app logger instance
     :return: redirect to login with flashed error message
     """
-    if 'user_id' not in session:
+    # The "cookie probe" check is performed here because the first point at which a typical session will suffer
+    # from cookies not working is when a CSRF token fails to validate. DM_COOKIE_PROBE_EXPECT_PRESENT controls whether
+    # we expect the probe cookie to be present at all - allowing the check to be rolled out and enabled gracefully
+    # among multiple frontends.
+    if current_app.config.get(
+        "DM_COOKIE_PROBE_EXPECT_PRESENT",
+        cookie_probe.DEFAULT_DM_COOKIE_PROBE_EXPECT_PRESENT
+    ) and (
+        request.cookies.get(
+            current_app.config.get("DM_COOKIE_PROBE_COOKIE_NAME", cookie_probe.DEFAULT_DM_COOKIE_PROBE_COOKIE_NAME)
+        ) != current_app.config.get("DM_COOKIE_PROBE_COOKIE_VALUE", cookie_probe.DEFAULT_DM_COOKIE_PROBE_COOKIE_VALUE)
+    ):
+        current_app.logger.info(
+            "cookie_probe.failed: cookies probably not working for this user"
+        )
+        return render_error_page(
+            status_code=400,
+            error_message="This feature requires cookies to be enabled for correct operation",
+        )
+    elif 'user_id' not in session:
```

Comment:
Rather than overloading the csrf handler completely can we break this out into a seperate function and then include that function in csrf handler based on the presence of the config value?

Reply:
Ahhi was on the verge of doing that yes, just couldn't decide what the call semantics should be... yeah I'll at least put the condition in a separate function.